packages/astro/src/core/build/static-build.ts

```diff
@@ -130,6 +70,17 @@ export async function staticBuild(opts: StaticBuildOptions) {
 	const ssrResult = (await ssrBuild(opts, internals, pageInput)) as RollupOutput;
 	info(opts.logging, 'build', dim(`Completed in ${getTimeStat(timer.ssr, performance.now())}.`));
 
+	const clientInput = new Set<string>([
+		...internals.discoveredHydratedComponents,
+		...internals.discoveredClientOnlyComponents,
+		...astroConfig._ctx.renderers.map(r => r.clientEntrypoint).filter(a => a) as string[],
+		...internals.discoveredScripts,
+	]);
```

Comment:
These are all of the client pieces to build; hydrated and client:only components, renderers, and hoisted scripts.